src/polyglot_ffi/generators/c_stubs_gen.py

```
@@ -188,28 +226,461 @@ def _convert_ocaml_to_c(self, return_type: IRType) -> list:
+            lines.append(f"    int* {param_name}_arr = (int*){param_name};")
+        elif element_type.name == "float":
+            lines.append(f"    double* {param_name}_arr = (double*){param_name};")
+        elif element_type.name == "bool":
+            lines.append(f"    int* {param_name}_arr = (int*){param_name};")
+
+        # Build OCaml list from C array in reverse order (for efficiency)
+        lines.append(f"    ml_{param_name} = Val_emptylist;  /* Start with empty list */")
+        lines.append(f"    for (int i = {param_name}_len - 1; i >= 0; i--) {{")
+
+        # Create cons cell
+        lines.append(f"        CAMLlocal1(cons);")
```

Ruff (0.14.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)